client/client.go

```diff
@@ -53,8 +59,20 @@ func NewClient(endpoint string) (Client, error) {
 		return nil, err
 	}
 
+	evmlogs, err := NewEvmLogsClient(endpoint)
```

Comment:
Why not reuse the raw `ethclient` returned by `NewEthClient`?

Reply:
Wanted a custom implementation with caching and also handled filtering so only logs relevant to a specific transaction hash are returned

Reply:
I guess what I mean is pass the `ethClient` created in `NewEthClient` into `NewEVMLogsClient` instead of creating another.